lib/mongo/auth/conversation_base.rb

```diff
@@ -47,6 +47,38 @@ def initialize(user, connection, **opts)
       def speculative_auth_document
         nil
       end
+
+      # @return [ Protocol::Message ] The message to send.
+      def build_message(connection, auth_source, selector)
```

Comment:
Nice refactoring introducing this.

Reply:
This was actually part of https://github.com/mongodb/mongo-ruby-driver/pull/2187, not this PR.